core/common/src/main/scala/org/typelevel/otel4s/Attributes.scala

```diff
@@ -121,6 +121,25 @@ sealed trait Attributes
 object Attributes extends SpecificIterableFactory[Attribute[_], Attributes] {
   private val Empty = new MapAttributes(Map.empty)
 
+  /** Allows creating [[Attributes]] from an arbitrary type `A`.
+    *
+    * @example
+    *   {{{
+    * case class User(id: Long, group: String)
+    *
+    * implicit val userAttributesMake: Attributes.Make[User] =
+    *   user => Attributes(Attribute("user.id", user.id), Attribute("user.group", user.group))
+    *
+    * val attributes: Attributes = Attributes.from(User(1L, "admin"))
+    *   }}}
+    *
+    * @tparam A
+    *   the type of the value
+    */
+  trait Make[A] {
+    def make(a: A): Attributes
```

Comment:
I think this is great; my only issue is that I think it would be nice to have a way to add the values directly to an `Attributes.Builder`. however, that would drastically complicate the API, and also introduce mutability concerns, so maybe this is the best solution as it is now

Reply:
Never thought of that. That would be a cool optimization. But it would complicate the API indeed :(